server/src/services/course.ts

```diff
@@ -10,3 +11,23 @@ export interface CourseWithTranslationAndInstance extends Course {
   CourseTranslations: Array<CourseTranslation>
   CourseInstances: Array<CourseInstance>
 }
+
+export async function findAllInstances(courseId: number): Promise<Array<CourseInstance>> {
+  const course: Array<Course> | null = await models.Course.findAll({
```

Comment:
`findOne` would be more appropriate, AFAIK `findAll` will continue to search through courses even when we have already found the one we want, whereas `findOne` will return the first (and only) found one right away.

Reply:
If we are querying here with the course PK, I would recommend using `findByPk`, then there is no need to use the `where` clause. It would also return just one Course instance. More detailed info [here](https://sequelize.org/docs/v6/core-concepts/model-querying-finders/#findbypk).

It could look something like this:
```Javascript
const course: Course | null = await models.Course.findByPk(courseId);

if (!course) throw new Error (`course with id ${courseId} not found`);
```

Reply:
True, that slipped my mind.